modules/ilmo/dns.tf

```diff
@@ -50,3 +50,88 @@ resource "azurerm_dns_txt_record" "ilmo_dmarc" {
     value = "v=DMARC1;p=none;sp=none;rua=mailto:dmarc@tietokilta.fi!10m;ruf=mailto:dmarc@tietokilta.fi!10m"
   }
 }
+
+
+terraform {
+  required_providers {
+    acme = {
+      source  = "vancluever/acme"
+      version = "2.19.0"
+    }
+  }
+}
+
+# A record for the web app
+resource "azurerm_dns_a_record" "tik-ilmo_a" {
+  name                = var.subdomain
+  resource_group_name = var.dns_resource_group_name
+  zone_name           = var.root_zone_name
+  ttl                 = 300
+  records             = data.dns_a_record_set.tik-ilmo_dns_fetch.addrs
+}
+
+# Azure verification key
+resource "azurerm_dns_txt_record" "tik-ilmo_asuid" {
+  name                = "asuid.${var.subdomain}"
+  resource_group_name = var.dns_resource_group_name
+  zone_name           = var.root_zone_name
+  ttl                 = 300
+
+  record {
+    value = azurerm_linux_web_app.ilmo_backend.custom_domain_verification_id
+  }
+}
+
+
+resource "azurerm_app_service_custom_hostname_binding" "tik-ilmo_hostname_binding" {
```

Comment:
I've tried to keep `dns.tf` for DNS entries only, and put everything else in `main.tf`